src/nifs/tests.rs

```diff
@@ -17,6 +19,37 @@ use crate::util::create_ro;
 
 use super::*;
 
+#[derive(thiserror::Error, Debug)]
+enum Error<C: CurveAffine> {
```

Comment:
Can we re-use the Error enum defined in upper scope instead of redefine it here?

Reply:
I don't understand you, this is a `Error` specifically for tests and it is the only one here. 

We can't reuse a `Error` from the main module for this one